src/ipc/ipc-zephyr.c

```diff
@@ -133,13 +133,24 @@ enum task_state ipc_platform_do_cmd(struct ipc *ipc)
 	/* perform command */
 	ipc_cmd(hdr);
 
-#if CONFIG_SOC_SERIES_INTEL_ADSP_CAVS
-	/* are we about to enter D3 ? */
-	if (ipc->pm_prepare_D3) {
-		/* no return - memory will be powered off and IPC sent */
+	if (ipc->task_mask & IPC_TASK_POWERDOWN) {
+#if defined(CONFIG_PM)
+		/**
+		 * @note For primary core this function
+		 * will only force set lower power state
+		 * in power management settings.
+		 * Core will enter D3 state after exit
+		 * IPC thread during idle.
+		 */
+		cpu_disable_core(PLATFORM_PRIMARY_CORE_ID);
+#else
+		/**
+		 * @note no return - memory will be
+		 * powered off and IPC sent.
+		 */
 		platform_pm_runtime_power_off();
+#endif /* CONFIG_PM */
```

Comment:
@aborisovich re the IPC timeouts and CI failures on TGL - IIUC the ```platform_pm_runtime_poweroff()``` is sending the IPC D3 complete response to the host. Do we know if ```cpu_disable_core()``` is doing the same ?
@kv2019i any thoughts why IPC D3 would timeout here (see CI results) ?

Reply:
@lgirdwood cpu_disable_core() should only be used if the custom SOF policy for Zephyr PM is used (see my comment above).